### PR TITLE
Add gkeonprem resources in google-beta

### PR DESCRIPTION
**Pull request by naitianliu-google**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:REPLACEME

```

### CODE REVIEW

**Review by rileykarson — COMMENTED** [review]
Chatted offline with @slevenick and it sounds like y'all are still iterating on the PR. Let me know when you want to start review. I'll warn you in advance, I'm likely to have you break up the change into a PR-per-resource- multi resource PRs take a long time to gather context on, and can take much longer to review.

Now that you're enrolled in the GCP org you can trigger generation by pushing new commits. gcbrun only works for repo maintainers, I think.  The current error is due to conflicting sources for https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/utils/gkeonprem_operation.go.erb- that one should be reproducible locally!

**Comment by naitianliu-google** [comment]
> Hmm- I'm not sure why that's happening. If you could push your current changes so I can see the generated code that would help!

Just pushed the current changes. Thanks for checking!

**Comment by naitianliu-google** [comment]
@rileykarson @slevenick Please help check why the error mentioned above is happening when you get a chance. Thanks!

**Comment by rileykarson** [comment]
I'm not entirely sure why that happened. Are you comfortable enough with Git to rebase on top of main? I'd recommend we do that and then try running again.

File: mmv1/products/gkeonprem/BareMetalCluster.yaml [hunk 1]
```diff
@@ -0,0 +1,997 @@
...
+    url_param_only: true
+    required: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: "adminClusterMembership"
+    required: true
+    description: |
+      The Admin Cluster this Bare Metal User Cluster belongs to.
+      This is the full resource name of the Admin Cluster's hub membership.
+    diff_suppress_func: projectNumberDiffSuppress
```
Comment (rileykarson) [inline mmv1/products/gkeonprem/BareMetalCluster.yaml:58]: This function changed to a capital, I think!

Comment (naitianliu-google) [reply mmv1/products/gkeonprem/BareMetalCluster.yaml:58]: > This function changed to a capital, I think!

Thanks for checking. Fixed, but still result in the `Value Conversion Error` (I have rebased on top of main) and followed the [guidance](https://googlecloudplatform.github.io/magic-modules/docs/how-to/add-mmv1-test/#tests-that-use-beta-features) adding `provider = google-beta` and `min_version: beta` accordingly. But still getting the error when running testacc command 

```
make testacc TEST=./google-beta TESTARGS='-run=TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample'
```

Error
```
==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -v -run=TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample
=== PAUSE TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample
=== CONT  TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample
    vcr_utils.go:144: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Invalid provider configuration
        
        Provider "registry.terraform.io/hashicorp/google-beta" requires explicit
        configuration. Add a provider block to the root module and configure the
        provider's required arguments as described in the provider documentation.
        
        
        Error: Value Conversion Error
        
          with provider["registry.terraform.io/hashicorp/google-beta"],
          on <empty> line 0:
          (source code not available)
        
        An unexpected error was encountered trying to convert tftypes.Value into
        google.ProviderModel. This is always an error in the provider. Please report
        the following to the provider developer:
        
        mismatch between struct and object: Object defines fields not found in
        struct: gkeonprem_custom_endpoint.
--- FAIL: TestAccgkeonpremVmwareCluster_gkeonpremVmwareClusterBasicExample (0.89s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-google-beta/google-beta 1.232s
FAIL
make: *** [GNUmakefile:15: testacc] Error 1
```

Is it possibly caused by local environments or anything I could try to locate this issue? @rileykarson @slevenick

**Pull request closed without merging**